common/src/main/java/io/netty5/util/internal/PlatformDependent.java

```diff
@@ -1019,6 +1020,17 @@ public static <C> Deque<C> newConcurrentDeque() {
         return new ConcurrentLinkedDeque<>();
     }
 
+    public static Object unwrapUnsafeOrNull() {
+        if (!hasUnsafe()) {
+            return null;
+        }
+        Class<?> callerClass = StackWalker.getInstance(Set.of(Option.RETAIN_CLASS_REFERENCE), 1).getCallerClass();
+        if (!callerClass.isAnnotationPresent(io.netty5.util.internal.UnsafeAccess.class)) {
```

Comment:
@normanmaurer I added this and the related annotation, to allow `UnsafeBuffer` to skip bounds checking, while also keep the stack height to a minimum to help inlining. The benchmark shows some effect from this.
Pointing this out since it now allows access to `Unsafe` outside of `PlatformDependent0`.